lib/marklogic.js

```diff
@@ -369,6 +371,42 @@ function MarkLogicClient(connectionParams) {
 
   this.logger = null;
 }
+
+/**
+ * Tests if a connection is successful.
+ * @method DatabaseClient#testConnection
+ * @since 2.1
+ * @returns true if the connection is successful and false otherwise with an error message
+ */
+MarkLogicClient.prototype.testConnection = function testConnection() {
+  var connectionParams = this.connectionParams;
+  var requestOptions = mlutil.copyProperties(connectionParams);
+  requestOptions.method = 'HEAD';
+  requestOptions.path = mlutil.databaseParam(connectionParams, '/v1/ping', '?');
+  var operation = new Operation(
+      'test operation', this, requestOptions, 'empty', 'single'
+      );
+  operation.statusCodeValidator = function testStatusCodes(statusCode, response){
+    if (statusCode >= 300)
+        this.responseStatusMessage = response.statusMessage;
+    return null;
+   };
+  operation.outputTransform = function addOutputTransform() {
+    var content = null;
+    if(this.responseStatusCode <300) {
+       content = {connected: true};
+    } else {
+       content = {connected: false,
+       httpStatusCode:  this.responseStatusCode,
+       httpStatusMessage: operation.responseStatusMessage
+       }
+    var wrapper = {content: [content]};
+    }
+   return wrapper;
```

Comment:
I've two concerns:

1. The wrapper variable is assigned only inside the else branch but returned from ddOutputTransform() in the if case as well.  Wouldn't the return value be undef in that case?
2.  What is the rationale for wrapping the content value instead of just returning the content value?